lib/rules/constructor-super.js

```diff
@@ -343,25 +325,30 @@ module.exports = {
              * @returns {void}
              */
             onCodePathSegmentLoop(fromSegment, toSegment) {
-                if (!(funcInfo && funcInfo.isConstructor && funcInfo.hasExtends)) {
+                if (!(funcInfo.isConstructor && funcInfo.hasExtends)) {
                     return;
                 }
 
-                // Update information inside of the loop.
-                const isRealLoop = toSegment.prevSegments.length >= 2;
-
                 funcInfo.codePath.traverseSegments(
                     { first: toSegment, last: fromSegment },
-                    segment => {
-                        const info = segInfoMap[segment.id] ?? new SegmentInfo();
+                    (segment, controller) => {
+                        const info = segInfoMap[segment.id];
+
+                        // skip segments after the loop
+                        if (!info) {
+                            controller.skip();
+                            return;
+                        }
+
                         const seenPrevSegments = segment.prevSegments.filter(hasSegmentBeenSeen);
+                        const calledInSomePreviousPaths = seenPrevSegments.some(isCalledInSomePath);
+                        const calledInEveryPreviousPaths = seenPrevSegments.every(isCalledInEveryPath);
 
-                        // Updates flags.
-                        info.calledInSomePaths = seenPrevSegments.some(isCalledInSomePath);
-                        info.calledInEveryPaths = seenPrevSegments.every(isCalledInEveryPath);
+                        info.calledInSomePaths ||= calledInSomePreviousPaths;
+                        info.calledInEveryPaths ||= calledInEveryPreviousPaths;
 
                         // If flags become true anew, reports the valid nodes.
-                        if (info.calledInSomePaths || isRealLoop) {
+                        if (calledInSomePreviousPaths) {
```

Comment:
This code that updates segments in loops was causing false positives. It didn't account for the case that the segment could have had a `super()` call inside the segment itself.